llvm/include/llvm/MC/MCAsmBackend.h

```diff
@@ -147,7 +147,8 @@ class MCAsmBackend {
   virtual void applyFixup(const MCAssembler &Asm, const MCFixup &Fixup,
                           const MCValue &Target, MutableArrayRef<char> Data,
                           uint64_t Value, bool IsResolved,
-                          const MCSubtargetInfo *STI) const = 0;
+                          const MCSubtargetInfo *STI,
+                          const MCFragment *Fragment) const = 0;
```

Comment:
This breaks all the targets that are not listed too. Maybe you can avoid changing that if you can store relocations and addend together?

Reply:
It doesn't matter if I store the Addends and the relocations together or separate, the issue is I cannot store the Addends until the fixup is applied which means I need to have the fragment, and if I store the Addend with the relocations, I will need the section in this function as well

Reply:
I think I am not gonna do the hack, but I understand what you mean